Cmake check for MPIIO does not use correct include directories.

The pass of the `DEFINITIONS` to the `check_macro()` for `MPIIO` within the `src/flib/CMakeLists.txt` does not include a `$` for the variable `MPIF_H_PATH`. Therefore it is not expanded as a variable and is instead just a string. 

```
-- Checking whether MPIIO is supported
check_macro: Change Dir: /home/tbrown/tmp/ParallelIO-pio2_3_1/build/src/flib/tryMPI_HAS_MPIIO/CMakeFiles/CMakeTmp

Run Build Command:"/bin/gmake" "cmTC_193e5/fast"
/bin/gmake -f CMakeFiles/cmTC_193e5.dir/build.make CMakeFiles/cmTC_193e5.dir/build
gmake[1]: Entering directory `/home/tbrown/tmp/ParallelIO-pio2_3_1/build/src/flib/tryMPI_HAS_MPIIO/CMakeFiles/CMakeTmp'
Building Fortran object CMakeFiles/cmTC_193e5.dir/TryMPIIO.f90.o
/opt/software/isv/intel/2017/compilers_and_libraries_2017.2.174/linux/bin/intel64/ifort   -I{MPIF_H_PATH} -c /home/tbrown/tmp/ParallelIO-pio2_3_1/build/CMake_Fortran_utils/TryMPIIO.f90 -o CMakeFiles/cmTC_193e5.dir/TryMPIIO.f90.o
/home/tbrown/tmp/ParallelIO-pio2_3_1/build/CMake_Fortran_utils/TryMPIIO.f90(2): error #5102: Cannot open include file 'mpif.h'
  include "mpif.h"
----------^
compilation aborted for /home/tbrown/tmp/ParallelIO-pio2_3_1/build/CMake_Fortran_utils/TryMPIIO.f90 (code 1)
gmake[1]: *** [CMakeFiles/cmTC_193e5.dir/TryMPIIO.f90.o] Error 1
gmake[1]: Leaving directory `/home/tbrown/tmp/ParallelIO-pio2_3_1/build/src/flib/tryMPI_HAS_MPIIO/CMakeFiles/CMakeTmp'
gmake: *** [cmTC_193e5/fast] Error 2

-- Checking whether MPIIO is supported - no
```

--- a/src/flib/CMakeLists.txt
+++ b/src/flib/CMakeLists.txt
@@ -161,7 +161,7 @@ if (MPIF_H_PATH)
   check_macro (MPI_HAS_MPIIO
     NAME TryMPIIO.f90
     HINTS ${CMAKE_MODULE_PATH}
-    DEFINITIONS -I{MPIF_H_PATH}
+    DEFINITIONS -I${MPIF_H_PATH}
     COMMENT "whether MPIIO is supported")
   if (${MPI_HAS_MPIIO})
     message (STATUS "MPIIO verified and enabled.")

